### DISCUSSION TITLE
Cluster restarted by Operator Suddenly

### DISCUSSION THREAD

**Discussion by bahaeddinoz** [strimzi/strimzi-kafka-operator#9696] [Q&A]
Hello,
I have a cluster on production with maintenance window and but operator restarted the kafka pods suddenly out of maintenance window which Tuesday at 5.58 AM. It was not on info mode which was not able to get reason exactly but It says because of certificates. When i check the certificates i can see they are still the old ones and didn't get renewed. (because generation date is same and they was valid for a longer time.)

**Log :** 
**Reconciliation #44038(timer) Kafka(strimzi-kafka/prod-kafka): Pod {prod-kafka-kafka-1} needs to be restarted. Reason: [clients-ca certificate removal]"**

**Maintenance Window**
  **Maintenance Time Windows:
    * * 6-7 ? * WED ***

I am not able share enough logs because it was on info mode but i just wanna ask about 2 things
- Why did not maintenance window stop operator ?
- It says renewal of certificates but nothing renewed or changed ?

Kafka Version: 3.2.3
Helm Version: 0.33.2

**Comment by scholzj** [comment]
I think it is hard to say what caused something without logs, configuration, understanding the previous state etc. In the only log message it also doesn't seem to say anything about renewal.

**Reply by bahaeddinoz** [reply]
Hi Jakub,
Thank your for the answer 🙏 
i thought it is related with certificate because of this  Reason: [clients-ca certificate removal]. Also i thought that it does not do any operation on non maintenance windows. That's why confused and i think the info logs can't help much to understand?

**Reply by scholzj** [reply]
Certificate removal would normally be related renewal. But without any context it is impossible to understand what exactly was happening in your cluster. I.e. was was there some certificate removal.
